pkg/middlewares/aws_bucket_policy_expander.go

```diff
@@ -44,25 +47,29 @@ func (m AwsBucketPolicyExpander) Execute(_, resourcesFromState *[]resource.Resou
 	return nil
 }
 
-func (m *AwsBucketPolicyExpander) handlePolicy(bucket *aws.AwsS3Bucket, results *[]resource.Resource) error {
-	if bucket.Policy == nil || *bucket.Policy == "" {
+func (m *AwsBucketPolicyExpander) handlePolicy(bucket *resource.AbstractResource, results *[]resource.Resource) error {
+	policyAttr, exist := bucket.Attrs.Get("policy")
+	if !exist || policyAttr == nil || policyAttr == "" {
 		return nil
 	}
+	policy := policyAttr.(string)
 
+	bucketAttr, _ := bucket.Attrs.Get("bucket")
+	bucketName := bucketAttr.(string)
 	data := map[string]interface{}{
-		"id":     bucket.Id,
-		"bucket": bucket.Bucket,
-		"policy": bucket.Policy,
+		"id":     bucket.TerraformId(),
+		"bucket": bucketName,
+		"policy": policy,
```

Comment:
We could totally write it that way since we won't need bucketName or policy var anymore when `S3BucketPolicy` will be refactored.

```
data := map[string]interface{}{
		"id":     bucket.TerraformId(),
		"bucket": (*bucket.Attrs)["bucket"],
		"policy": policyAttr,
	}
```